internal/dbdrivers/redis.go

```diff
@@ -444,22 +549,28 @@ func (clients *RedisDBConn) Set(c context.Context, key string, data interface{},
 	return nil
 }
 
-func (clients *RedisDBConn) HSet(c context.Context, key string, field string, data interface{}, ttl time.Duration) error {
-	jsonBytes, err := json.Marshal(data)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-
-	if err := clients.Primary.HSet(c, key, field, jsonBytes).Err(); err != nil {
-		return errors.WithStack(err)
-	}
-
-	if ttl > 0 {
-		if err := clients.Primary.Expire(c, key, ttl).Err(); err != nil {
+// - HSet("myhash", map[string]interface{}{"key1": "value1", "key2": "value2"})
+// - HSet("myhash", []string{"key1", "value1", "key2", "value2"})
+// - HSet("myhash", "key1", "value1", "key2", "value2")
+func (clients *RedisDBConn) HSet(c context.Context, key string, args ...interface{}) error {
+	switch args[0].(type) {
+	case map[string]interface{}:
+		fieldWithValuesMap := args[0]
+		if err := clients.Primary.HSet(c, key, fieldWithValuesMap).Err(); err != nil {
+			return errors.WithStack(err)
+		}
+	case []string:
+		fieldWithValuesArray := args[0]
+		if err := clients.Primary.HSet(c, key, fieldWithValuesArray).Err(); err != nil {
 			return errors.WithStack(err)
 		}
+	case string:
+		if err := clients.Primary.HSet(c, key, args).Err(); err != nil {
```

Comment:
I think should be ` if err := clients.Primary.HSet(c, key, args...).Err(); err != nil {`